1-Draft/RFCNNNN-Simplify-Module-Manifests.md

```diff
@@ -0,0 +1,170 @@
+---
+RFC: RFCnnnn
+Author: Kirk Munro
+Status: Draft
+SupercededBy: 
+Version: 1.0
+Area: Module Manifests
+Comments Due: July 15, 2019
+Plan to implement: Yes
+---
+
+# Title
+
+When PowerShell module support first came out in PowerShell 2.0, module manifests had a specific list of keys that could be used to define the manifest. Additional keys were added in PowerShell 3.0, but that hurt the end user experience because at the time, if you loaded a module with a manifest containing keys that were not recognized/supported in your version of PowerShell, PowerShell would return an unhelpful error about the module keys being invalid instead of checking for the minimum required version of PowerShell and returning an error indicating that the module only supports that version or later.
+
+Beyond PowerShell 3.0, keys started being added to a PSData subsection of the PrivateData area of modules. You can see these keys in use by the PowerShell Gallery. That approach has its own challenges, because now keys are defined in a hierarchy, which adds complexity and should not be necessary.
+
+Fast forward to today, and things have changed enough that we can reconsider how we set up a module manifest. `Import-Module` has been checking the required version of PowerShell before validating key names for several years now (at least since PowerShell 5.1).  Earlier versions of Windows PowerShell are no longer on mainstream support.
+
+With all of this in mind, I propose that we flatten the module manifest structure, reading values for keys at the root first, and then if they are not set at the root, looking in the PSData section under PrivateData for backwards compatibility support. Future keys added to the manifest as part of PowerShell should all be added to the root, leaving `PrivateData` for 3rd party or user information in a manifest, like it was intended.
+
+## Motivation
+
+    As a scripter,
+    I can generate manifests using `New-ModuleManifest` with all keys stored at the root of the hashtable,
+    so that I can work with module manifests more easily.
+
+## User Experience
+
+Here's an example showing how `New-ModuleManifest` would work after this change:
```

Comment:
For greater clarity we could add commented `PSData` section with comment that these keys is moved out.
We could implement this to inform users about the change.

Reply:
Added a note to alternate proposals about this.